### PR TITLE
EDRD Phase-1 March release Package

### PR DESCRIPTION
**Pull request by deepakmulamalla**
Contains the changes related to 2nd generation package creation & fixes related to EDRD Phase-1 March release Package

### CODE REVIEW

File: force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls [hunk 1]
```diff
@@ -32,4 +34,27 @@ public with sharing class CarePgmEnrolleeTriggerHandler {
...
+/**
+* @author: Deepak
+* @date: 11/07/2024
+* @description: The purpose of this method is update the Account Ref. No based on CPE Name
+* @Modification Log: [Date] - [Change Reference] - [Changed By] - [Description]
+*/ 
+    public static void updateEnrolleeNameOnAcc(List<CareProgramEnrollee> carePgmEnrolleeList){
+        List<Account> accList = new List<Account>();
+        String cPEDRD = 'EDRD';
+        List<CareProgramEnrollee> cPEList =[SELECT Id, CareProgram.Name FROM CareProgramEnrollee
```
Comment (yashrohan-cgi) [inline force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls:47]: This should be part of selector class

File: force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls [hunk 2]
```diff
@@ -32,4 +34,27 @@ public with sharing class CarePgmEnrolleeTriggerHandler {
...
+    
+/**
+* @author: Deepak
+* @date: 11/07/2024
+* @description: The purpose of this method is update the Account Ref. No based on CPE Name
+* @Modification Log: [Date] - [Change Reference] - [Changed By] - [Description]
+*/ 
+    public static void updateEnrolleeNameOnAcc(List<CareProgramEnrollee> carePgmEnrolleeList){
+        List<Account> accList = new List<Account>();
+        String cPEDRD = 'EDRD';
```
Comment (yashrohan-cgi) [inline force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls:46]: Add to constants or labels

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls [hunk 3]
```diff
@@ -132,14 +134,16 @@ public with sharing class ESA_cls_caseTriggerHandler {
...
             Boolean isProrityChanged = oldCaseMap.get(caseObj.id).Priority != newCaseMap.get(caseObj.id).Priority;
             Boolean isTypeChanged = oldCaseMap.get(caseObj.id).Type != newCaseMap.get(caseObj.id).Type;
-            if(caseObj.RecordTypeId == recordTypeCaseEDRD && (isStatusChanged || isProrityChanged || isTypeChanged)){
+            Boolean isEDRDRefChanged = oldCaseMap.get(caseObj.id).EDRD_Ref_No__c != newCaseMap.get(caseObj.id).EDRD_Ref_No__c;
+            if(caseObj.RecordTypeId == recordTypeCaseEDRD && (isStatusChanged || isProrityChanged || isTypeChanged || isEDRDRefChanged)){
                 caseIdSet.add(caseObj.Id);
             }
         }
-        for(MedicationRequest MRObj:[SELECT Id, Case__r.Type, Case__r.Status, Case__r.Priority FROM MedicationRequest WHERE Case__c IN: caseIdSet]){
+        for(MedicationRequest MRObj:[SELECT Id, Case__r.Type, Case__r.Status, Case__r.Priority, Case__r.EDRD_Ref_No__c FROM MedicationRequest WHERE Case__c IN: caseIdSet]){
```
Comment (yashrohan-cgi) [inline force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:142]: Add query in selector class

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls [hunk 4]
```diff
@@ -289,4 +293,39 @@ public with sharing class ESA_cls_caseTriggerHandler {
             System.debug('Exception caught: ' + err.getMessage());
```
Comment (yashrohan-cgi) [inline force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:293]: remove debug log

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls [hunk 5]
```diff
@@ -289,4 +293,39 @@ public with sharing class ESA_cls_caseTriggerHandler {
...
+        
+        for(Case caseObj: newCaseList){
+            Boolean isAccountIdChanged = trigger.isUpdate && newCaseMap.get(caseObj.Id).accountId != oldCaseMap.get(caseObj.Id).accountId;
+            if(!String.isEmpty(caseObj.AccountId) && caseObj.RecordTypeId == recordTypeCaseEDRD && 
+               (trigger.isInsert || isAccountIdChanged)){
+                   patientIdVsCase.put(caseObj.AccountId, caseObj);
+               }
+        }
+        if(!patientIdVsCase.isEmpty()){
+            List<Account> accList = [SELECT Id, Name, (SELECT Id, Name, CareProgramId FROM AccountCareProgramEnrollees
```
Comment (yashrohan-cgi) [inline force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:315]: add in selector class

File: force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls [hunk 6]
```diff
@@ -32,4 +34,27 @@ public with sharing class CarePgmEnrolleeTriggerHandler {
...
+                                           WHERE Id IN: carePgmEnrolleeList];
+        
+        for(CareProgramEnrollee carePgmEnrollee: carePgmEnrolleeList){
+            if(cPEDRD.equalsIgnoreCase(carePgmEnrollee.CareProgram.Name)){
+                accList.add(new Account(Id = carePgmEnrollee.AccountId, EDRD_Ref_No__pc = carePgmEnrollee.Name));
+            }
+        }  
+        
+        if(!accList.isEmpty()){
+            Update accList;
```
Comment (yashrohan-cgi) [inline force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls:57]: Do we have log object , we can add try catch to catch the exception and store exception in log record if available.

File: force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls (hunk 1 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls:46]: @yashrohan-cgi , we do not have a frame setup for selector class.

File: force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls (hunk 2 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls]: @yashrohan-cgi , I've constant to query.

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls (hunk 3 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:142]: @yashrohan-cgi , we do not have a frame setup for selector class.

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls (hunk 4 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:293]: it is try and catch block, debug statement is used to capture the exception message.

File: force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls (hunk 6 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/CarePgmEnrolleeTriggerHandler.cls:54]: No we do not have any object to capture exceptions

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls (hunk 4 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:293]: @yashrohan-cgi , we do not have a frame setup for selector class.

File: force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls (hunk 5 above)
Comment (deepakmulamalla) [reply force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls:315]: @yashrohan-cgi , we do not have a frame setup for selector class.

**Review by yashrohan-cgi — APPROVED** [review]

**Review by NataliaNikishina — APPROVED** [review]
@deepakmulamalla PR composition was validated. Thank you!

**Review by RanadheerRG — APPROVED** [review]
@deepakmulamalla  Approved

**Pull request merged**